spring-kafka/src/main/java/org/springframework/kafka/listener/FallbackBatchErrorHandler.java

```diff
@@ -54,6 +56,8 @@ class FallbackBatchErrorHandler extends KafkaExceptionLogLevelAware
 
 	private boolean ackAfterHandle = true;
 
+	private boolean retrying;
```

Comment:
`volatile`?

Reply:
No need; always used from the consumer thread.

Reply:
So, you mean that `onPartitionsAssigned()` is called withing that `ErrorHandlingUtils.retryBatch()` process?
Otherwise I don't see how this variable is involved...

Thanks

Reply:
`ErrorHandlingUtils.retryBatch` pauses the consumer and polls in a while loop until retries are exhausted, and then resumes.

We need to repause the consumer if `onPartitionsAssigned()` is called while we are in that loop (it is called from `poll()` in the kafka-clients).

So the thread that tests this field is the same thread that is looping within `retryBatch`.